yarn-project/canary/src/uniswap_trade_on_l1_from_l2.test.ts

```diff
@@ -212,134 +267,159 @@ describe.skip('uniswap_trade_on_l1_from_l2', () => {
       value: parseEther('1'),
     });
 
-    const meBeforeBalance = await wethContract.read.balanceOf([ethAccount.toString()]);
+    const wethL1BeforeBalance = await wethContract.read.balanceOf([ownerEthAddress.toString()]);
     // 1. Approve weth to be bridged
     await wethContract.write.approve([wethTokenPortalAddress.toString(), wethAmountToBridge], {} as any);
 
     // 2. Deposit weth into the portal and move to L2
     // generate secret
-    const secret = Fr.random();
-    const secretHash = await computeMessageSecretHash(secret);
-    const secretString = `0x${secretHash.toBuffer().toString('hex')}` as `0x${string}`;
-    const deadline = 2 ** 32 - 1; // max uint32 - 1
+    const [secretForMintingWeth, secretHashForMintingWeth] = await generateClaimSecret();
+    const [secretForRedeemingWeth, secretHashForRedeemingWeth] = await generateClaimSecret();
     logger('Sending messages to L1 portal');
-    const args = [owner.toString(), wethAmountToBridge, deadline, secretString, ethAccount.toString()] as const;
-    const { result: messageKeyHex } = await wethTokenPortal.simulate.depositToAztecPublic(args, {
-      account: ethAccount.toString(),
+    const args = [
+      wethAmountToBridge,
+      deadline,
+      secretHashForMintingWeth.toString(true),
+      secretHashForRedeemingWeth.toString(true),
+      ownerEthAddress.toString(),
+    ] as const;
+    const { result: messageKeyHex } = await wethTokenPortal.simulate.depositToAztecPrivate(args, {
+      account: ownerEthAddress.toString(),
     } as any);
     await wethTokenPortal.write.depositToAztecPublic(args, {} as any);
 
-    const currentL1Balance = await wethContract.read.balanceOf([ethAccount.toString()]);
-    logger(`Initial Balance: ${currentL1Balance}. Should be: ${meBeforeBalance - wethAmountToBridge}`);
-    expect(currentL1Balance).toBe(meBeforeBalance - wethAmountToBridge);
+    const currentL1Balance = await wethContract.read.balanceOf([ownerEthAddress.toString()]);
+    logger(`Initial Balance: ${currentL1Balance}. Should be: ${wethL1BeforeBalance - wethAmountToBridge}`);
+    expect(currentL1Balance).toBe(wethL1BeforeBalance - wethAmountToBridge);
     const messageKey = Fr.fromString(messageKeyHex);
 
     // Wait for the archiver to process the message
     await sleep(5000);
     // send a transfer tx to force through rollup with the message included
-    const transferAmount = 1n;
-    await transferWethOnL2(wethL2Contract, owner, receiver, transferAmount);
+    const transferAmount = 0n;
```

Comment:
Seems unnecessary to have a variable to account for 0 here, when used later you will also just be subtracting zero so think we could live without it.

Reply:
that's a good point - will remove. Left over from last code